src/pickley/__init__.py

```diff
@@ -665,6 +666,16 @@ def program_version(path, logger=None):
             if r.succeeded:
                 return PickleyConfig.parsed_version(r.output or r.error)
 
+    @staticmethod
+    def installed_sibling_folders(canonical_name):
+        """Sibling installations of the form '<canonical_name>-<version>'."""
+        regex = re.compile(r"^(.+)-(\d+[.\d+]+)$")
+        for item in runez.ls_dir(CFG.meta):
+            if item.is_dir():
+                m = regex.match(item.name)
```

Comment:
you can use `.fullmatch` here and you don't have to wrap with `^...$` (it's safer too, more like `^(?:...)$` which is a little clunky to read